net/src/mock_worker.rs

```diff
@@ -284,17 +279,24 @@ impl NetWorker for MockWorker {
     /// we got a message from holochain core
     /// forward to our mock singleton
     fn receive(&mut self, data: Protocol) -> NetResult<()> {
-        let mut mock = get_mock()?;
-
         if let Ok(wrap) = ProtocolWrapper::try_from(&data) {
             if let ProtocolWrapper::TrackApp(app) = wrap {
+                // get a write lock first
+                let mut map_lock = MOCK_MAP.write().unwrap();
+                let mut mock = map_lock
+                    .entry(self.network_name.clone())
+                    .or_insert_with(|| Mutex::new(MockSystem::new()))
+                    .lock()
+                    .unwrap();
+
                 let (tx, rx) = mpsc::channel();
                 self.mock_msgs.push(rx);
                 mock.register(&app.dna_address, &app.agent_id, tx)?;
                 return Ok(());
             }
         }
-
+        let map_lock = MOCK_MAP.read().unwrap();
+        let mut mock = map_lock.get(&self.network_name).unwrap().lock().unwrap();
```

Comment:
I don't believe the "get" is guaranteed to work so I'd put an expect instead of unwrap here.